### PR TITLE
cli: Use correct type annotations

### PR DESCRIPTION
**Pull request by mariano54**
These ports are Optionals but they were incorrectly types as `int`.
Also removes unused argument `harvester_rpc_port`, since we connect to farmer now

### CODE REVIEW

File: chia/cmds/wallet.py [hunk 1]
```diff
@@ -97,11 +99,11 @@ def send_cmd(
     default=None,
 )
 @click.option("-f", "--fingerprint", help="Set the fingerprint to specify which wallet to use", type=int)
-def show_cmd(wallet_rpc_port: int, fingerprint: int) -> None:
+def show_cmd(wallet_rpc_port: Optional[int], farmer_rpc_port: int, fingerprint: int) -> None:
```
Comment (xdustinface) [inline chia/cmds/wallet.py:102]: Why are you adding `farmer_rpc_port` here?

File: chia/cmds/wallet.py [hunk 2]
```diff
@@ -97,11 +99,11 @@ def send_cmd(
...
     import asyncio
     from .wallet_funcs import execute_with_wallet, print_balances
 
-    asyncio.run(execute_with_wallet(wallet_rpc_port, fingerprint, {}, print_balances))
+    asyncio.run(execute_with_wallet(wallet_rpc_port, fingerprint, {"farmer_rpc_port": farmer_rpc_port}, print_balances))
```
Comment (xdustinface) [inline chia/cmds/wallet.py:106]: And here

**Review by xdustinface — CHANGES_REQUESTED** [review]
See the question in comments + i would personally split this PR into two (or at least change the title to reflect the PRs full intention).. one which fixes type annotations and one which drops unused parameters since there is also an unused `--wallet-rpc-port` in `show_cmd` of `show.py`.. maybe there are more if you dig deeper :smile:

File: chia/cmds/wallet.py (hunk 1 above)
Comment (mariano54) [reply chia/cmds/wallet.py:102]: Not sure, i think that is a mistake. I'll revert

**Comment by mariano54** [comment]
Unrelated to this PR, but I think it might be better to remove all RPC arguments as they clutter the code, and user can just change it in the config. 

Otherwise change it to the root chia command instead of having it on each sub command.
I think this RPC argument is mostly leftover code from the alpha releases of the CLI that just got carried over, but not very useful

**Review by xdustinface — APPROVED** [review]

**Pull request merged**